sympy/core/evalf.py

```diff
@@ -600,7 +600,8 @@ def evalf_add(v: 'Add', prec: int, options: OPT_DICT) -> TMP_RES:
         options['maxprec'] = min(oldmaxprec, 2*prec)
 
         terms = [evalf(arg, prec + 10, options) for arg in v.args]
-        n = terms.count(S.ComplexInfinity)
+        # remove any None so they won't be sympified during comparison
+        n = [term for term in terms if not isinstance(term, tuple)].count(S.ComplexInfinity)
```

Comment:
```suggestion
        n = sum(1 for i in terms if i is S.ComplexInfinity)
```

Reply:
Here's a more direct workaround.

Reply:
I considered `is S.ComplexInfinity` too, but I wonder if there are any cases that we do want to catch where a term in `terms` evaluates to `S.ComplexInfinity`

Reply:
You mean an element in `term`?

Reply:
It doesn't look like `zoo` will be in a tuple (based on the use of `ComplexInfinity` in evalf.py). In addition the docstring says:
```
    An optional tuple, ``(re, im, re_acc, im_acc)``
    which are the real, imaginary, real accuracy
    and imaginary accuracy respectively. ``re`` is
    an mpf value tuple and so is ``im``. ``re_acc``
    and ``im_acc`` are ints.
    NB: all these return values can be ``None``.
    If all values are ``None``, then that represents 0.
    Note that 0 is also represented as ``fzero = (0, 0, 0, 0)``.

```
It should also note that `zoo` will possibly be returned. I don't know why it says an 'optional tuple' unless I missed something. It returns `r`, possibly the result after `chop_parts` so it should be a tuple or `zoo`.

Reply:
I see. Then yes, this would solve the problem. However, it essentially introduces an extra check because comparisons between `tuple`s and `Basic`s are bugged. So that should be fixed rather than introducing a check here.

We should definately keep this as a backup solution if it turns out that solving the underlying problem is not (at this point) possible.